### PR TITLE
:lock: security allow actuator probes

### LINKED ISSUE #37: Test security configuration
**Is your feature request related to a problem? Please describe.**
Currently the security configuration isn't tested. As this is one of the crucial parts it should be tested very well.

**Describe the solution you'd like**

Test for security configuration which especially tests all specific matchers (i.e. actuator, public routes, ...)

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Permit actuator probe endpoints for deployment health checks.

Required for #23 

Closes #37

### CODE REVIEW

**Comment by DanielOber** [comment]
LGTM and works locally. But can you write tests for it like we did [here](https://github.com/it-at-m/dave-backend/blob/sprint/src/test/java/de/muenchen/dave/configuration/SecuredWebAppTest.java)

File: refarch-gateway/src/main/java/de.muenchen.oss.refarch.gateway/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -78,6 +78,8 @@ public SecurityWebFilterChain springSecurityFilterChain(ServerHttpSecurity http)
                             .pathMatchers(LOGOUT_SUCCESS_URL).permitAll()
                             .pathMatchers("/api/*/info",
                                     "/actuator/health",
+                                    "/actuator/health/liveness",
+                                    "/actuator/health/readiness",
```
Comment (devtobi) [inline refarch-gateway/src/main/java/de.muenchen.oss.refarch.gateway/configuration/SecurityConfiguration.java:82]: Would be `/actuator/*` be easier to manage? Or is this to loosely defined?

Comment (simonhir) [reply refarch-gateway/src/main/java/de.muenchen.oss.refarch.gateway/configuration/SecurityConfiguration.java:82]: Thought as restrictive as possible makes sense here. I would leave it like this for the moment as also every other projects defines the matchers without wildcards.
Also I already created #36 to hide the whole actuator endpoint from the outside.

**Comment by simonhir** [comment]
> LGTM and works locally. But can you write tests for it like we did [here](https://github.com/it-at-m/Wahllokalsystem/blob/dev/wls-eai-service/src/test/java/de/muenchen/oss/wahllokalsystem/eaiservice/configuration/SecurityConfigurationTest.java)

Unfortunately, this was not straight forward to implement because of context problems, so I have extracted it to #37

File: refarch-gateway/src/test/java/de.muenchen.oss.refarch.gateway/configuration/SecurityConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+@AutoConfigureWebTestClient
+@AutoConfigureObservability
+@ActiveProfiles(profiles = { SPRING_TEST_PROFILE })
+public class SecurityConfigurationTest {
+    @Autowired
+    WebTestClient api;
+
+    @Test
+    void accessSecuredResourceRootThenUnauthorized() {
+        // api.get().uri("/").exchange().expectStatus().isUnauthorized();
```
Comment (DanielOber) [inline refarch-gateway/src/test/java/de.muenchen.oss.refarch.gateway/configuration/SecurityConfigurationTest.java:23]: remove unused code

**Review by DanielOber — APPROVED** [review]
LGTM

**Pull request merged**